packages/@lwc/wds-core/src/browser/wireMockUtil.js

```diff
@@ -0,0 +1,189 @@
+class TestWireAdapterTemplate {
+  static initialValue;
+
+  static emit(value, filterFn) {
+    TestWireAdapterTemplate.initialValue = value;
+    let instances = Array.from(TestWireAdapterTemplate._wireInstances);
+    if (typeof filterFn === 'function') {
+      instances = instances.filter((instance) => filterFn(instance.getConfig()));
+    }
+    for (const instance of instances) {
+      instance.emit(value);
+    }
+  }
+
+  static getLastConfig() {
+    return TestWireAdapterTemplate._lastConfig;
+  }
+
+  constructor(dataCallback) {
+    this.config = {};
+    this._dataCallback = dataCallback;
+    TestWireAdapterTemplate._wireInstances.add(this);
+  }
+
+  update(config) {
+    this.config = config;
+    TestWireAdapterTemplate._lastConfig = config;
+  }
+
+  connect() {
+    if (TestWireAdapterTemplate.initialValue !== undefined) {
+      this.emit(TestWireAdapterTemplate.initialValue);
+    }
+    TestWireAdapterTemplate._lastConfig = {};
+    TestWireAdapterTemplate._wireInstances.add(this);
+  }
+
+  disconnect() {
+    TestWireAdapterTemplate._wireInstances.delete(this);
+  }
+
+  emit(value) {
+    this._dataCallback(value);
+  }
+
+  getConfig() {
+    return this.config;
+  }
+}
+
+TestWireAdapterTemplate._lastConfig = null;
+TestWireAdapterTemplate._wireInstances = new Set();
+
+function buildTestWireAdapter() {
+  const _a = class TestWireAdapter extends TestWireAdapterTemplate {};
+  _a._lastConfig = null;
+  _a._wireInstances = new Set();
+  return _a;
```

Comment:
This is where I think the `initialValue` will need to be attached to the class. `_a` is unique to the particular wire mock that is created, so attaching the value here should work fine and prevent initial value collisions between multiple mocks in the same test.